RocketChat/Rocket.Chat
# [FIX] Discussion not updating rooms list and not checking right permissions

## Proposed changes

Validate if a user can send message on the channel he/she is trying to create the discussion.

Also update rooms list when a discussion is created.

## Issue(s)

Closes https://github.com/RocketChat/Rocket.Chat/issues/15430

## How to test or reproduce

* Mute an user on a room, with that user, create a discussion to that room
* After creating a discussion check on other user's rooms list and the room will not be updated.

## Screenshots

## Types of changes




## Checklist



## Changelog




## Further comments
